### PR TITLE
CHECKOUT-3852 Expose proper error type for Coupon/GiftCertificates

### PR DESCRIPTION
**Pull request by capsula4**
## What?
As per title

## Why?
So consumers can access more information like why coupon failed

@bigcommerce/checkout

### CODE REVIEW

File: src/coupon/coupon-selector.ts [hunk 1]
```diff
@@ -13,11 +14,11 @@ export default class CouponSelector {
         return this._coupon.data;
     }
 
-    getRemoveError(): Error | undefined {
+    getRemoveError(): RequestError | undefined {
```
Comment (davidchin) [inline src/coupon/coupon-selector.ts]: Do we need to specify the error response type?

File: src/checkout/checkout-store-error-selector.ts [hunk 2]
```diff
@@ -330,7 +332,7 @@ export default class CheckoutStoreErrorSelector {
      *
      * @returns The error object if unable to apply, otherwise undefined.
      */
-    getApplyCouponError(): Error | undefined {
+    getApplyCouponError(): RequestError<StorefrontErrorResponseBody> | undefined {
```
Comment (davidchin) [inline src/checkout/checkout-store-error-selector.ts]: Correct me if I'm wrong, but I think these getters return more than just `RequestError`. Any error that's thrown inside an observable action will be surfaced through here. So we might still need to have `Error` or `CustomError`.

Comment (capsula4) [reply src/checkout/checkout-store-error-selector.ts]: at the moment this error is generated in the couponActionCreator, when the promise from couponRequestSender fails, so I think it's safe to assume it's RequestError in this case.

File: src/coupon/coupon-selector.ts (hunk 1 above)
Comment (capsula4) [reply src/coupon/coupon-selector.ts]: Actually I think we might not want to. The generic type is the body of the original error response, but part of the work I did was to standardise how we return RequestErrors from the 3 different APIs that we interact with (internal, payments, storefront), so we shouldn't really need to access the original API payload (even though it's available)

**Pull request merged**